core/src/test/java/org/apache/iceberg/TestRowDelta.java

```diff
@@ -1519,6 +1574,69 @@ public void testConcurrentDeletesRewriteSameDeleteFile() {
         statuses(Status.ADDED));
   }
 
+  @TestTemplate
+  public void testConcurrentManifestRewriteWithDeleteFileRemoval() throws IOException {
+    // Manifest rewrite isn't supported on branches currently
+    assumeThat(branch).isEqualTo("main");
```

Comment:
Had an old PR to fix this https://github.com/apache/iceberg/pull/10265/files

for now though i don't think we need to block on that, because I think the test is useful as it is